OpenRA.Mods.Common/Traits/BotModules/Squads/States/AirStates.cs

```diff
@@ -220,12 +177,13 @@ public void Tick(Squad owner)
 				if (BusyAttack(a))
 					continue;
 
-				if (!ReloadsAutomatically(a))
+				var ammoPools = a.TraitsImplementing<AmmoPool>();
+				if (!ReloadsAutomatically(ammoPools, a.TraitOrDefault<Rearmable>()))
```

Comment:
I must have missed this during all the back and forth. Can someone elaborate why we're suddenly doing the trait lookups outside of the method (= high coupling)? Since we're not `ToArray`ing or `ToList`ing we'll still enumerate twice worst case, so there should basically be no difference between doing the lookups here and doing it inside the method (except coupling).

Reply:
ummmm.. yeah I also said readbility is down here in exchange of the performance.

Reply:
But it will require more thinking on probably change all those functions back for readbility and low coupling, since they are all about the ammo related.

Reply:
My point was that this doesn't look like it gives us any performance gains.

Reply:
The idea was that both `ReloadsAutomatically` and `HasAmmo()` below were looking up `a.TraitsImplementing<AmmoPool>();` before, so I was assuming that caching it once, passing it through as argument and then running a foreach on them would save at least one of the (up to) two look-ups as well as the additional enumerations.

I'm not that firm on the topic though, if you say passing them as array would be more efficient we should do that, of course.

Reply:
Changed to `ToArray`